packages/documentation/src/pages/getting-started/common-tasks/new-end-to-end-test.mdx

```diff
@@ -0,0 +1,153 @@
+    .assert.title('My Application Title | Veterans Affairs')
+    .click('.schemaform-start-button');
+
+  // disable scrolling
+  FormsTestHelpers.overrideFormsScrolling(client);
+
+  // assert navigation is successful
+  E2eHelpers.expectNavigateAwayFrom(client, '/introduction');
+
+  // Personal Information page.
+  client.expect.element('input[name="root_veteranFullName_first"]').to.be
+    .visible;
+  // use functions from the helper file to perform all actions on the page
+  MyApplicationHelpers.completePersonalInformation(client, testData.data);
+  client.axeCheck('.main').click('.form-panel .usa-button-primary');
+  E2eHelpers.expectNavigateAwayFrom(
+    client,
+    '/my-application/personal-information',
+  );
+  ```
+
+**my-application-helpers.js**
+```
+const mock = require('platform/testing/e2e/mock-helpers');
+const Timeouts = require('platform/testing/e2e/timeouts.js');
+const Auth = require('platform/testing/e2e/auth.js');
+const moment = require('moment');
+
+function completePersonalInformation(client, data) {
+  client
+    .waitForElementVisible(
+      'input[name="root_veteranFullName_first"]',
+      Timeouts.normal,
+    )
+    .fill(
+      'input[name="root_veteranFullName_first"]',
+      data.veteranFullName.first,
+    )
+    .fill('input[name="root_veteranFullName_last"]', data.veteranFullName.last)
+}
+```
+_These are recommendations not requirements._
+
+- separate navigation from field input
+  - use a **main test file** for navigation, assertions, and calls helpers
+  - use a **helper file** that for filling out forms
+- create separate, numbered **main test files** to organize tests by their focus
+  - **00-all-fields.e2e.spec.js** - required and optional fields
+  - **01-required.e2e.spec.js** - only required fields
+  - **02-accessibility.e2e.spec.js** - validates accessibility
+  - **03-auth.e2e.spec.js** - validates authentication
+  - **04-cross-cutting-feature.e2e.spec.js** - validates one feature used across several pages (e.g. save in progress)
+- export the end to end test using `E2eHelpers.createE2eTest()`- see [Helpers](#helpers)
+- group tests by pages and use a comment to indicate what page is being tested
+- mock all api responses before starting the test. See [Mocking API responses](#mockimg-api-responses)
+- use `waitForElementVisible` before interacting with any element on the page
+- use `Timeouts` constants for setting timeouts (`platform/testing/e2e/timeouts.js`)
+- use helpers for filling data and performing actions on the page
+- perform `axeCheck` on the main body of the application on each page- see [Custom Nightwatch commands](#custom-nightwatch-commands)
+- assert that each navigation is successful
+
+## Mocking API responses
+
+**my-application-helpers.js**
+
+A mock server runs with the end to end tests to allow tests to make production-like calls.
+
+```
+const mock = require('platform/testing/e2e/mock-helpers');
+
+function initApplicationSubmitMock() {
+  mock(null, {
```

Comment:
Maybe out of scope for these docs, but sometimes we need to pass a user token as the first arg to `mock()`. Also, even more likely out of scope, I've seen some examples of calling `client.mockData()` to mock a failure response for an API call, which is something the `mock()` helper doesn't appear to support.

Reply:
It looks like the token is just added to the 2nd argument- @jbalboni is there a reason why we have this first argument and not just have people add it as `token:` property to the response object?

Reply:
I don't think so. Here is an example of the token getting passed to the `mock()` call:

https://github.com/department-of-veterans-affairs/vets-website/blob/2cb102ae260833573905e033b67d93939254da4b/src/applications/hca/tests/hca-helpers.js#L380-L392

And that function gets called with a token here: https://github.com/department-of-veterans-affairs/vets-website/blob/26f7edca07ee0b40fb7bc3df29c46ed5f2ffee50/src/platform/forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js#L9

Reply:
Yeah, I don't see why it's written this way.

Reply:
Also, I wonder if we should suggest using the `client.mockData` helper instead of using `mock` directly? The interface is less confusing (token is an optional second param) and it will actually wait for the mock http call to finish before moving on with the test.